### ISSUE TITLE
Either "Loaded Modules" or "Runtime Class Lookups" crash the Unveil app on unc0ver

### ISSUE THREAD

**Issue by yhnmj6666** [jjolano/shadow#48]
Hello jjolano, thank you very much for developing such a great jailbreak detection bypass tweak. 
I'm testing shadow on my iPhone against the Unveil app, but the app insta-crash once I enable `Loaded Modules` or `Runtime Class Lookups`. If I keep both options off, lots of modifications are shown as detected. 
I've attach the screenshot of Unveil and the crash logs here. If you have any time please kindly check these things.

crash logs: 
[unveil.zip](https://github.com/jjolano/shadow/files/10029184/unveil.zip)

screenshots: 
<img src="https://user-images.githubusercontent.com/15858973/202390697-e4339b8f-0ee5-4f7c-ba25-714ba7f0cdf7.png" width=40% height=40%>

Thanks again for your hard work!

**Comment by jjolano** [comment]
I had no idea this app existed. Thanks!

**Comment by jjolano** [comment]
@yhnmj6666 could you check if version 3.1.1 fixes the crash issue? At least for Loaded Modules hook. Unfortunately, I believe Runtime Class Lookups crash is related to substitute.

**Comment by yhnmj6666** [comment]
Thank you for the new versions! 
Unfortunately `Loaded modules` is still crashing Unveil on unc0ver, as well as `Runtime Class Lookups`. And the new option `Enforce sandbox` is crashing the app as well.
Comparing to the previous version, the injected libraries are hided but one hooked libraries still remains. (/usr/lib/dyld)

crashlogs:
[unveil.zip](https://github.com/jjolano/shadow/files/10069031/unveil.zip)

screenshot of the app:
<img src="https://user-images.githubusercontent.com/15858973/203362329-2fb718a8-f3e1-4176-befe-ac51ad443e85.jpg" width=40% height=40%>

**Comment by jjolano** [comment]
Thanks again for the logs. Seems to be substitute related since all logs appear to have an abort call - maybe something isn't getting hooked successfully. Unfortunately, I do not yet have a device using unc0ver/substitute so no possibility of debugging on my end.

I wonder if substitute puts logs somewhere...

**Comment by yhnmj6666** [comment]
I just found that substitute write the logs to `/tmp/cydia.log` and the log lines start with `SubstituteLog:`. But the log lines when I'm playing with Unveil is missing, not sure what happened. 
The only thing it logs looks like this:
```
Attempted to hook non-existant selector "_openWithAppLink:state:XPCConnection:completionHandler:" in class "LSAppLink"
```
I'll try again later and post logs here if I can grab the log.

ref: https://github.dev/sbingner/substitute/blob/788722b2338ca50d0751985fdb069b0d41460225/lib/substitute-internal.h#L12

**Comment by jjolano** [comment]
I have updated Shadow to version 3.3, with some new hooking code specifically for substitute. Runtime Class Lookups will likely still have issues, but hopefully Loaded Modules would work fine now.

**Comment by jjolano** [comment]
The crashes should be fixed now with 3.3.3
